scripts/i18n-keysets-sort.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Weblate create PR with changed and approved keysets with sorted keys
+ * That's why we need to decrease diff
+ */
+const {spawnSync} = require('child_process');
+const {readFileSync, writeFileSync} = require('fs');
+const changedUncommittedFiles = spawnSync('git', ['status', '-s']);
+
+const needSortFiles =
+    changedUncommittedFiles.output?.[1] && changedUncommittedFiles.output?.[1].length;
+if (!needSortFiles) {
+    process.exitCode = 0;
+}
+const filesList = changedUncommittedFiles.output[1].toString();
+const filteredFiles = filesList
+    .split('\n')
+    .filter((item) => {
+        const hasKeysetFiles = item.trim().match(/(A|M)\s*(src\/i18n-keysets\/).*\.json/);
```

Comment:
You can add a hook for specific paths in [lintstagedrc.json](https://github.com/datalens-tech/datalens-ui/blob/main/.lintstagedrc.json) and then it will not provoke the script to run on unnecessary files.